.github/workflows/test-telemetry-setup.yaml

```
@@ -1,51 +1,93 @@
+        uses: rapidsai/shared-actions/telemetry-dispatch-load-base-env-vars@telemetry-dispatch-actions
         with:
-            ref: ${{env.SHARED_ACTIONS_REF}}
-            path: ./shared-actions
-      # Run job with traceparent. We'll validate that this matches in the summary data.
-      - name: Get job traceparent in matrix job
-        uses: ./shared-actions/telemetry-traceparent
-        id: job-traceparent
-      - name: Generate traceparent for a step
-        uses: ./shared-actions/telemetry-traceparent
-        id: step-traceparent
+            load_service_name: "true"
+      - name: Check if service name took on an unexpected value
+        run: |
+          echo "(should be the value set to the OTEL_SERVICE_NAME env var in base-env-setup job)"
+          [ "${OTEL_SERVICE_NAME}" = "${{needs.base-env-setup.outputs.service-name}}" ] || exit 1
+      - name: Telemetry summarize
+        uses: rapidsai/shared-actions/telemetry-dispatch-write-summary@telemetry-dispatch-actions
         with:
-            step_name: "Download gha-tools with git clone"
-      - name: Echo computed step traceparent
+          cert_concat: "${{ secrets.OTEL_EXPORTER_OTLP_CA_CERTIFICATE }};${{ secrets.OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE }};${{ secrets.OTEL_EXPORTER_OTLP_CLIENT_KEY }}"
+
+      - name: Check if service name was altered during telemetry summary
         run: |
-            echo "GHA tools clone job traceparent: ${{ steps.step-traceparent.outputs.traceparent }}"
-      - name: Test OTel export of job JSON
-        uses: ./shared-actions/telemetry-summarize
+          echo "(should be the value set to the OTEL_SERVICE_NAME env var in base-env-setup job)"
+          [ "${OTEL_SERVICE_NAME}" = "${{needs.base-env-setup.outputs.service-name}}" ] || exit 1
+      - name: Query the Tempo HTTP API and check that our trace is present and has expected properties
+        run: |
+          TRACE_ID=$( cut -d '-' -f 2 <<< "$TRACEPARENT" );
+          echo "Trace ID is: ${TRACE_ID}";
+          TRACE_URL="${OTEL_EXPORTER_OTLP_ENDPOINT/4318/3200}/api/traces/${TRACE_ID}"
+          echo "Trace URL is: ${TRACE_URL}"
+          curl \
+            --cert /tmp/certs/client.crt.pem --key /tmp/certs/client.key.pem --cacert /tmp/certs/ca.crt.pem \
+            -Gs "${TRACE_URL}" > trace_record.json;
+      - name: Upload trace record
+        uses: actions/upload-artifact@v4
         with:
-            traceparent: ${{ steps.job-traceparent.outputs.traceparent}}
+            name: trace-record
+            path: trace_record.json
+      - name: Validate span metadata
+        # these are not returned in any particular order. The span kind is the only one
+        # that we can reliably expect to be the same.
+        run: |
+          span_kind="$(jq -r '.batches[0].scopeSpans[0].spans[0].kind' trace_record.json )";
+          echo "Checking if span kind is as expected"
+          echo "Span kind is: "${span_kind}""
```

Does this need escaping?
```suggestion
          echo "Span kind is: \"${span_kind}\""
```